src/test/java/org/example/stepsDefinitions/LoginFunctionalitiesSteps.java

```diff
@@ -0,0 +1,30 @@
+package org.example.stepsDefinitions;
+
+import io.cucumber.java.en.*;
+
+
+public class LoginFunctionalitiesSteps {
+	@Given("the user has navigated to the login page")
+	public void theUserHasNavigatedToTheLoginPage() {
+
+	}
+
+	@When("the user enters the correct username and password")
+	public void theUserEntersTheCorrectUsernameAndPassword() {
+
+	}
+
+	@Then("the user should be redirected to the home page")
+	public void theUserShouldBeRedirectedToTheHomePage() {
+		
+	}
+
+	@When("the user enters an incorrect username or password")
+	public void theUserEntersAnIncorrectUsernameOrPassword() {
```

Comment:
This is redundant. If you will have one common step for login form then all you need it just describe the test name properly and you do not need the step for user form with incorrect data.
I.E.
test name: Error message appears when login with incorrect data.
steps:
given -> The 'web-service' lunch in browser
and -> user enter username: incorect and password: incorrect to the login form
and -> user click on 'login' button
then -> the home page opens/displays.

something like that.

Reply:
Correct Method Implemented with incorrect data